src/phantasm-renderer/common/single_cache.hh

```diff
@@ -62,16 +63,22 @@ public:
     void cull_all(gpu_epoch_t current_gpu_epoch, F&& destroy_func)
     {
         auto lg = std::lock_guard(_mutex);
-        auto f_can_cull = [&](map_element const& elem) { return elem.num_references == 0 && elem.required_gpu_epoch <= current_gpu_epoch; };
+        auto f_can_cull = [&](map_element const& elem) -> bool { return elem.num_references == 0 && elem.required_gpu_epoch <= current_gpu_epoch; };
+
+        cc::vector<cc::hash_t> keys_to_remove;
+        keys_to_remove.reserve(_map.size());
```

Comment:
doesn't this reserve over-alloc a lot if only a few elements are deleted?

Reply:
This function is only ever triggered by an explicit user call to Context::clear_XY_cache, it's not the cull function for normal operation (which is still missing), so on average this is intended to remove most if not all keys.